[Imraj Singh] iP

imrajsinghsandhu opened the pull request:
<h1>DukePro </h1>

>“Your mind is for having ideas, not holding them.” – David Allen([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST*  to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

imrajsinghsandhu committed:
latest

imrajsinghsandhu committed:
level 1

imrajsinghsandhu committed:
Level 2

imrajsinghsandhu committed:
level 3

imrajsinghsandhu committed:
Level 4

imrajsinghsandhu committed:
Level-5

imrajsinghsandhu committed:
Level-6

imrajsinghsandhu committed:
A-TextUiTesting

imrajsinghsandhu committed:
level 7

imrajsinghsandhu committed:
Level-8

imrajsinghsandhu committed:
A-MoreOOP

imrajsinghsandhu committed:
A-Packages

imrajsinghsandhu committed:
A-JUnit

imrajsinghsandhu committed:
A-Jar

imrajsinghsandhu committed:
A-JavaDoc

imrajsinghsandhu committed:
A-CodingStandard

imrajsinghsandhu committed:
Level-9

Harish-coding commented on src/main/java/Chat.java:
```diff
@@ -0,0 +1,97 @@
+import java.util.*;
```
maybe the import statement could explicitly mention that ArrayList class and scanner class are used.

Harish-coding commented on src/main/java/Chat.java:
```diff
@@ -0,0 +1,97 @@
+
+            String input = sc.nextLine();
+
+            // we will check from 0 to the first space
```
the description of the code in comments makes it easier to understand what the code is doing! 👍

Harish-coding commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,46 @@
+     * 
+     * @return
+     */
+    public String getStatusIcon() {
+        return (this.isDone ? "X" : " "); // mark done task with X
+    }
+
+    // Getters
+    public boolean getIsDone() {
+        return this.isDone;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
```
The use of verbs in the method names makes it clear what the method is doing... but maybe adding javadocs comment to describe the methods, would make it even more easier, I guess.... what do u think?

Harish-coding reviewed (commented):
Looks good to me, just some minor issues to fix 👍.

imrajsinghsandhu committed:
update level 4

imrajsinghsandhu committed:
update the test files

benmurphyy commented on src/main/java/Events.java:
```diff
@@ -0,0 +1,19 @@
+ */
+public class Events extends Task {
+
+    private String at;
```
Is this variable name for the date/time that the event occurs too short? Perhaps make it a bit longer and more explicit like dateOfEvent ?

benmurphyy commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,45 @@
+ */
+public class Task {
+    private String description;
+    private boolean isDone;
```
Good naming of this variable, it is clear that it is a boolean variable just from the name! 👍

benmurphyy commented on src/main/java/Chat.java:
```diff
@@ -0,0 +1,127 @@
+        ArrayList<Task> tasks = new ArrayList<Task>();
+
+        // takes in the incoming prompt
+        Scanner sc = new Scanner(System.in);
```
Perhaps give this scanner a lengthier name other than sc? So anyone reading the code below will not get confused what sc is...maybe call it scanner instead which is clearer?

imrajsinghsandhu committed:
rename Chat.java to Duke.java

benmurphyy commented on src/main/java/Chat.java:
```diff
@@ -0,0 +1,127 @@
+            LINE_BREAK;
+    public static String LINE_INTRO = "Nah, here's your list";
+    public static String MARK_MESSAGE = "Power la Mr Bosssssss, mark alr bro!";
+    public static String UNMARK_MESSAGE = "No probs bro, unmarked already!";
```
I like that you wanted set this reused phrases as constants! Perhaps consider adding the `final` attribute to them to make it clear that they are constants, and prevent modification?

benmurphyy commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,22 @@
+ */
+public class Deadlines extends Task {
+
+    private String by;
```
Perhaps this variable name for the date/time for the deadline too short? Perhaps make it a bit longer and more explicit like dateOfDeadline?

benmurphyy reviewed (commented):
I found your code quite easy to read and understand and mostly adhering to code quality guidelines! Just a few recommendations regarding the naming of variables!

imrajsinghsandhu committed:
update to level 5

imrajsinghsandhu committed:
update level-6

imrajsinghsandhu committed:
delete older exception classes

imrajsinghsandhu committed:
delete the wild card imports

imrajsinghsandhu committed:
Update Duke.java

imrajsinghsandhu committed:
branch-Level-7 edits

imrajsinghsandhu committed:
branch-Level-8

potential bugs at toString method of Deadline and Event classes

imrajsinghsandhu committed:
Merge branch 'branch-Level-8'

imrajsinghsandhu committed:
update A-MoreOOP

updated the files into moreOOP format

imrajsinghsandhu committed:
Create Storage.java

imrajsinghsandhu committed:
latest updates to Storage

imrajsinghsandhu committed:
quick update to date time formatting

imrajsinghsandhu committed:
Add boolean isDate method

imrajsinghsandhu committed:
Update duke.txt

imrajsinghsandhu committed:
Update Duke.java

imrajsinghsandhu committed:
Update Duke.java

imrajsinghsandhu committed:
Level-8

imrajsinghsandhu committed:
Organise into packages

imrajsinghsandhu committed:
A-JUnit

imrajsinghsandhu committed:
A-JavaDoc

imrajsinghsandhu committed:
A-CodingStandard

imrajsinghsandhu committed:
Level-9

imrajsinghsandhu committed:
A-Gradle

imrajsinghsandhu committed:
Level-10

imrajsinghsandhu committed:
update tests

imrajsinghsandhu committed:
Level-10

imrajsinghsandhu committed:
Update Duke.java

imrajsinghsandhu committed:
Eliminate unnecessary comments

imrajsinghsandhu committed:
Add assert to Storage constructor

Error occurs up when the filepath is wrongly declared, and the task attribute is not updated upon calling of load() method in Storage class

Need to assert the correct filepath so the task attribute gets correctly updated

Add an assert declaration to check for the correct filepath, allowing us to ensure the filepaths is correct, and the task attribute updated correctly

imrajsinghsandhu committed:
A-CodeQuality

Command attributes classes: extract a command class Commands

Command attribute classes have common behaviours

The common behaviours across command classes cause code duplication

Extracting the common behaviours into a super class allows us to use polymorphism when dealing with command attributes.

Lets pull up attributes common to all command attributes classes into a new parent class Commands.

imrajsinghsandhu committed:
Merge pull request #2 from imrajsinghsandhu/branch-A-Assertions

Add assert to Storage constructor

imrajsinghsandhu committed:
Merge pull request #3 from imrajsinghsandhu/branch-A-CodeQuality

A-CodeQuality

imrajsinghsandhu committed:
Add a Tag command

We are now able to tag our tasks, and the tags will show up when we initiate the "list" command

imrajsinghsandhu committed:
improved formatting for Parser.java

imrajsinghsandhu committed:
Update duke.txt

imrajsinghsandhu committed:
Create Ui.png

imrajsinghsandhu committed:
Update README.md

imrajsinghsandhu committed:
Finalised changes to Duke

imrajsinghsandhu committed:
final commits for Duke

imrajsinghsandhu committed:
Merge branch 'branch-Level-8'

imrajsinghsandhu committed:
Merge branch 'branch-Level-7'

imrajsinghsandhu committed:
Delete ParseDukeTxt.class

imrajsinghsandhu committed:
Revert "final commits for Duke"

This reverts commit 5f58536285dba499595452583e9a9eb17f4eab6f.

imrajsinghsandhu committed:
Revert "Revert "final commits for Duke""

This reverts commit 89aef69890046efdae61d155f11701229f49ab62.

imrajsinghsandhu committed:
Edit Parser and delete dead code

imrajsinghsandhu committed:
Code quality improvements

Remove repetitive code in Deadline file, and make other code quality edits to keep in line with checkstyle

imrajsinghsandhu committed:
Update README.md

Add a UI picture of Duke to README

imrajsinghsandhu committed:
Update README.md

imrajsinghsandhu committed:
Update README.md

imrajsinghsandhu committed:
Update duke.txt

imrajsinghsandhu committed:
Final Updates

imrajsinghsandhu committed:
Update duke.txt
